arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -354,6 +360,40 @@ public void setBatchSize(int batchSize) {
     }
   }
 
+  private static final class PositionVectorReader extends VectorizedArrowReader {
+    private long rowStart;
+
+    @Override
+    public VectorHolder read(VectorHolder reuse, int numValsToRead) {
+      Field arrowField = ArrowSchemaUtil.convert(MetadataColumns.ROW_POSITION);
+      FieldVector vec = arrowField.createVector(ArrowAllocation.rootAllocator());
+      ((BigIntVector) vec).allocateNew(numValsToRead);
```

Comment:
Can this follow an approach similar to `VectorizedArrowReader` and not create a new `FieldVector` and `NullabilityHolder` for every invocation?
https://github.com/apache/iceberg/blob/52f44687a533f0d9b9895c9dc4326b7ef7785654/arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java#L123-L125

Reply:
Make sense to me, updated in https://github.com/apache/iceberg/pull/1356/commits/73de3696204c8fe1ed99cfddda2cf488c413f735